storagemarket/provider_loop.go

```diff
@@ -150,6 +155,15 @@ func (p *Provider) loop() {
 
 			dealReq.rsp <- acceptDealResp{&api.ProviderDealRejectionInfo{Accepted: true}, nil}
 
+		case storageSpaceDealReq := <-p.storageSpaceChan:
+			deal := storageSpaceDealReq.deal
+			if err := p.storageManager.Untag(p.ctx, deal.DealUuid); err != nil && !xerrors.Is(err, db.ErrNotFound) {
+				p.dealLogger.LogError(deal.DealUuid, "failed to untag storage space", err)
+			} else {
+				p.dealLogger.Infow(deal.DealUuid, "untagged storage space")
+			}
+			storageSpaceDealReq.done <- struct{}{}
```

Comment:
I suggest just closing the channel so you don't need to worry about blocking the main loop
```suggestion
			close(storageSpaceDealReq.done)
```